flight/ODLC/point_finder.py

```diff
@@ -0,0 +1,143 @@
+from shapely.geometry import Point, Polygon
+from shapely.ops import nearest_points
+import utm
+
+
+def latlon_to_utm(coords: Dict[str, float]) -> Dict[str, float]:
+    """Converts latlon coordinates to utm coordinates and adds the data to the dictionary
+
+    Parameters
+    ----------
+    coords : Dict[str, float]
+        A dictionary containing lat long coordinates
+
+    Returns
+    -------
+    Dict[str, float]
+        An updated dictionary with additional keys and values with utm data
+    """
+
+    utm_coords = utm.from_latlon(coords["latitude"], coords["longitude"])
+    coords["utm_x"] = utm_coords[0]
+    coords["utm_y"] = utm_coords[1]
+    coords["utm_zone_number"] = utm_coords[2]
+    coords["utm_zone_letter"] = utm_coords[3]
+    return coords
+
+
+def all_latlon_to_utm(list_of_coords: List[Dict[str, float]]) -> List[Dict[str, float]]:
+    """Converts a list of dictionarys with latlon data to add utm data
+
+    Parameters
+    ----------
+    list_of_coords : List[Dict[str, float]]
+        A list of dictionaries that contain lat long data
+
+    Returns
+    -------
+    List[Dict[str, float]]
+        list[dict]: An updated list of dictionaries with added utm data
+    """
+
+    for i, _ in enumerate(list_of_coords):
+        list_of_coords[i] = latlon_to_utm(list_of_coords[i])
+    return list_of_coords
+
+
+def scale_polygon(my_polygon: Polygon, scale_factor: float = 0.1) -> Polygon:
+    """Scale a shapely polygon by a percentage amount
+
+    Parameters
+    ----------
+    my_polygon : Polygon
+        Polygon that will be scaled
+    scale_factor : float, optional
+        Amount the polygon will be scaled, by default 0.1
+
+    Returns
+    -------
+    Polygon
+        Scale a shapely polygon by a percentage amount
+    """
+
+    x_s = list(my_polygon.exterior.coords.xy[0])
+    y_s = list(my_polygon.exterior.coords.xy[1])
+    x_center = 0.5 * min(x_s) + 0.5 * max(x_s)
+    y_center = 0.5 * min(y_s) + 0.5 * max(y_s)
+    min_corner = Point(min(x_s), min(y_s))
+    center = Point(x_center, y_center)
+    shrink_distance = center.distance(min_corner) * scale_factor
+    my_polygon_resized = my_polygon.buffer(-shrink_distance)
+
+    return my_polygon_resized
+
+
+def find_closest_point(
+    odlc: Dict[str, float],
+    boundary_points: List[Dict[str, float]],
+    obstacles: List[Dict[str, float]],
+) -> Tuple[Dict[str, float], List[float]]:
+    """Finds the closest safe point to the ODLC while staying within the flight boundary
+
+    Parameters
+    ----------
+    odlc : Dict[str, float]
+        Point data for the ODLC object
+    boundary_points : List[Dict[str, float]]
+        Point data which makes up the flight boundary
+    obstacles : List[Dict[str, float]]
+        Point data for the obstacles
+
+    Returns
+    -------
+    Tuple[Dict[str, float], List[float]]
+        Closest safe point, and the shrunken boundary (for plotting)
+    """
+
+    poly_points = []
```

Comment:
If you can change this initialization to a list comprehension method rather than iteration I would.